mopidy_mpd/protocol/connection.py

```diff
@@ -51,3 +52,50 @@ def ping(context):
         Does nothing but return ``OK``.
     """
     pass
+
+
+@protocol.commands.add("tagtypes")
+def tagtypes(context, *parameters):
+    """
+    *mpd.readthedocs.io, connection settings section:*
+
+        ``tagtypes``
+
+        Shows a list of available song metadata.
+
+        ``tagtypes disable {NAME...}``
+
+        Remove one or more tags from the list of tag types the client is interested in.
+
+        ``tagtypes enable {NAME...}``
+
+        Re-enable one or more tags from the list of tag types for this client.
+
+        ``tagtypes clear``
+
+        Clear the list of tag types this client is interested in.
+
+        ``tagtypes all``
+
+        Announce that this client is interested in all tag types.
+    """
+    parameters = list(parameters)
+    if parameters:
+        subcommand = parameters.pop(0).lower()
+        if subcommand not in ("all", "clear", "disable", "enable"):
+            raise exceptions.MpdArgError("Unknown sub command")
+        elif subcommand == "all":
+            context.session.tagtypes.update(tagtype_list.TAGTYPE_LIST)
+        elif subcommand == "clear":
+            context.session.tagtypes.clear()
+        elif subcommand == "disable":
+            context.session.tagtypes.difference_update(parameters)
+        elif subcommand == "enable":
+            enabled_types = [
+                value
+                for value in parameters
+                if value in tagtype_list.TAGTYPE_LIST
+            ]
+            context.session.tagtypes.update(enabled_types)
```

Comment:
I think we should align our implementation with mpd, and raise an error for unknown tagtypes.

```suggestion
            if not all((value in tagtype_list.TAGTYPE_LIST for value in parameters)):
                raise exceptions.MpdArgError("Unknown tag type")
            context.session.tagtypes.update(parameters)
```

Reply:
I changed it to raise an error for invalid tagtypes, but it will still just do nothing if no tagtypes are specified.

Reply:
oh, i didn't think about that! what do you think about adding this before the `if not all` check?

```
if not parameters:
    raise exceptions.MpdArgError("Not enough arguments")
```

Reply:
it would probably be best to put both checks into a small function to reduce duplication